### PR TITLE
Add PVC management

### PR DESCRIPTION
**Pull request by ruivieira**
Refers:
- [RHOAIENG-14581](https://issues.redhat.com/browse/RHOAIENG-14581)
- https://github.com/trustyai-explainability/trustyai-service-operator/issues/325

This PR adds support for PVC has output storage for LM-Eval jobs. It supports two modes, managed PVC and existing PVC.

#### Managed PVC

A PVC can be specified in the `LMEvalJob` as

```yaml
apiVersion: trustyai.opendatahub.io/v1alpha1
kind: LMEvalJob
spec:
  outputs:
    pvcManaged:
      size: 5Gi
```

The PVC will be created and managed by the operator, named `<LMEvalJob name>-pvc`. The only supported configuration at the moment is `size`.

#### Existing PVC

An existing PVC can be passed to the LM-Eval job using a PVC name, e.g.

```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: my-pvc
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 1Gi
```

```yaml
apiVersion: trustyai.opendatahub.io/v1alpha1
kind: LMEvalJob
spec:
  outputs:
    pvcName: my-pvc
```

This PVC will not be managed by the operator.
If both managed and existing PVC are specified, the operator will prefer the managed configuration and ignore the existing PVC one.

### CODE REVIEW

File: controllers/lmes/storage.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+// handleManagedPVC handles a managed PVC for the LMEvalJob
+func (r *LMEvalJobReconciler) handleManagedPVC(ctx context.Context, log logr.Logger, job *lmesv1alpha1.LMEvalJob) error {
+	pvcName := generateManagedPVCName(job)
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	err := r.Get(ctx, types.NamespacedName{Name: pvcName, Namespace: job.Namespace}, pvc)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("PVC for LMEvalJob " + job.Name + "not found. Creating.")
+			managedPVC := generateMagedPVCResource(job)
```
Comment (christinaexyou) [inline controllers/lmes/storage.go:61]: fix typo - should be `managedPVCResource`

**Review by christinaexyou — APPROVED** [review]

File: controllers/lmes/storage.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	return job.Name + "-pvc"
+}
+
+// generateMagedPVCResource creates a PersistentVolumeClaim for an LMEvalJob
+func generateMagedPVCResource(job *lmesv1alpha1.LMEvalJob) *corev1.PersistentVolumeClaim {
+	pvcName := generateManagedPVCName(job)
+	pvc := corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pvcName,
+			Namespace: job.Namespace,
```
Comment (yhwang) [inline controllers/lmes/storage.go:27]: nit: should there be an `OwnerReference` pointing back to the LMEvalJob?

Comment (ruivieira) [reply controllers/lmes/storage.go:27]: Hi @yhwang, `generateMagedPVCResource` would solely generate the managed PVC resource. This PVC definition will always be created with `createPVC` which sets the job as owner with `ctrl.SetControllerReference(job, pvc, r.Scheme)`. Does that fulfill the requirement? I can change it if not.

Comment (yhwang) [reply controllers/lmes/storage.go:27]: you're right. I missed the code in `createPVC()` function.

**Pull request merged**